keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/memory/InMemoryUnhappyVetoRepository.kt

```diff
@@ -28,42 +28,45 @@ class InMemoryUnhappyVetoRepository(
 
   private val resources: MutableMap<String, Record> = mutableMapOf()
 
-  override fun markUnhappyForWaitingTime(resourceId: String, application: String, wait: Duration) {
+  override fun markUnhappyForWaitingTime(resourceId: String, application: String, wait: Duration?) {
     resources[resourceId] = Record(application, calculateExpirationTime(wait))
   }
 
   override fun delete(resourceId: String) {
     resources.remove(resourceId)
   }
 
-  override fun getOrCreateVetoStatus(resourceId: String, application: String, wait: Duration): UnhappyVetoStatus {
+  override fun getOrCreateVetoStatus(resourceId: String, application: String, wait: Duration?): UnhappyVetoStatus {
     val record = resources[resourceId]
     if (record == null) {
       resources[resourceId] = Record(application, calculateExpirationTime(wait))
       return UnhappyVetoStatus(shouldSkip = true, shouldRecheck = false)
     }
     return UnhappyVetoStatus(
-      shouldSkip = record.recheckTime > clock.instant(),
-      shouldRecheck = record.recheckTime < clock.instant()
+      shouldSkip = record.recheckTime.shouldSkip(),
+      shouldRecheck = record.recheckTime.shouldRecheck()
     )
   }
 
   override fun getAll(): Set<String> {
     val now = clock.instant()
-    return resources.filter { it.value.recheckTime > now }.keys.toSet()
+    return resources.filter { it.value.recheckTime.shouldSkip(now) }.keys.toSet()
```

Comment:
This seemed backwards to me at first sight in my he context of this particular function. It’d be easier to read if the extension was named something like `isNullOrAfter()`, IMHO.

Reply:
Same thing with `shouldRecheck()`. I’d go with `isBefore()`, which even matches the existing function name/behavior in non-nullable `Instant`.